tools/Fabric.Mcp.Tools.PublicApi/src/FabricPublicApiSetup.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Azure.Mcp.Core.Areas;
+using Azure.Mcp.Core.Commands;
+using Fabric.Mcp.Tools.PublicApi.Commands.BestPractices;
+using Fabric.Mcp.Tools.PublicApi.Commands.PublicApis;
+using Fabric.Mcp.Tools.PublicApi.Services;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Fabric.Mcp.Tools.PublicApi;
+
+public class FabricPublicApiSetup : IAreaSetup
+{
+    public string Name => "publicapis";
+
+    public void ConfigureServices(IServiceCollection services)
+    {
+        services.AddSingleton<IResourceProviderService, EmbeddedResourceProviderService>();
+        services.AddSingleton<IFabricPublicApiService, FabricPublicApiService>();
+        services.AddHttpClient<FabricPublicApiService>();
+    }
+
+    public void RegisterCommands(CommandGroup rootGroup, ILoggerFactory loggerFactory)
+    {
+        var fabricPublicApis = new CommandGroup(Name,
+            """
+            Microsoft Fabric Public API Access - Retrieve OpenAPI specifications and example files 
+            from Microsoft Fabric's official API documentation. Use this tool when you need to:
+            - Build applications that integrate with Microsoft Fabric APIs
+            - Understand API endpoints, request/response schemas, and authentication requirements
+            - Generate client SDKs or API wrappers for Fabric services
+            - Review example API calls and responses for development guidance
+            - Explore available APIs across different Fabric workloads
+            This tool provides read-only access to Microsoft Fabric's public API documentation 
+            repository on GitHub. It does NOT interact with live Fabric resources or require 
+            authentication - it only retrieves API specifications and examples.
+            """);
+        rootGroup.AddSubGroup(fabricPublicApis);
+
+        // Create Fabric subgroups
+        var bestPractices = new CommandGroup("bestpractices", "API Examples and Best Practices - Commands for retrieving example API request/response files and implementation guidance from Microsoft Fabric's documentation repository.");
+        fabricPublicApis.AddSubGroup(bestPractices);
+
+        fabricPublicApis.AddCommand("workloads list", new DiscoverPublicWorkloadsCommand(loggerFactory.CreateLogger<DiscoverPublicWorkloadsCommand>()));
```

Comment:
@jongio do you know if we support for commands like this? 

Its one way to remove another layer of groups I guess